test/functional/Java12andUp/src/org/openj9/test/java_lang_invoke/Test_MethodHandle.java

```diff
@@ -0,0 +1,164 @@
+	}
+
+	/*
+	 * Test Java 12 API MethodHandle.describeConstable() for field types
+	 */
+	@Test(groups = { "level.sanity" })
+	public void testMethodHandleDescribeConstableField() {
+		testMethodDescribeConstableGeneral("testMethodHandleDescribeConstableField for getter", getterHandle);
+		testMethodDescribeConstableGeneral("testMethodHandleDescribeConstableField for setter", setterHandle);
+		testMethodDescribeConstableGeneral("testMethodHandleDescribeConstableField for static_getter", staticGetterHandle);
+		testMethodDescribeConstableGeneral("testMethodHandleDescribeConstableField for static_setter", staticSetterHandle);
+	}
+
+	/*
+	 * Test Java 12 API MethodHandle.describeConstable() for indirect handle types
+	 */
+	@Test(groups = { "level.sanity" })
+	public void testMethodHandleDescribeConstableIndirect() {
+		testMethodDescribeConstableGeneral("testMethodHandleDescribeConstableIndirect for virtual", virtualHandle);
+		testMethodDescribeConstableGeneral("testMethodHandleDescribeConstableIndirect for interface_virtual", interfaceVirtualHandle);
+	}
+
+	/*
+	 * Test Java 12 API MethodHandle.describeConstable() for direct handle types
+	 */
+	@Test(groups = { "level.sanity" })
+	public void testMethodHandleDescribeConstableDirect() {
+		testMethodDescribeConstableGeneral("testMethodHandleDescribeConstableDirect for static", staticHandle);
+		testMethodDescribeConstableGeneral("testMethodHandleDescribeConstableDirect for interface_static", interfaceStaticHandle);
+		testMethodDescribeConstableSpecial("testMethodHandleDescribeConstableDirect for special", specialHandle);
+		testMethodDescribeConstableSpecial("testMethodHandleDescribeConstableDirect for interface_special", interfaceSpecialHandle);
+	}
+	
+	private void testMethodDescribeConstableGeneral(String testName, MethodHandle handle) {
+		testMethodDescribeConstableGeneralSub(testName, handle, lookup);
+	}
+
+	private void testMethodDescribeConstableSpecial(String testName, MethodHandle handle) {
+		testMethodDescribeConstableGeneralSub(testName, handle, specialLookup);
+	}
+
+	private void testMethodDescribeConstableGeneralSub(String testName, MethodHandle handle, MethodHandles.Lookup resolveLookup) {
+		Optional<MethodHandleDesc> handleDescOp = handle.describeConstable();
+
+		if (handleDescOp.isPresent()) {
+			MethodHandleDesc handleDesc = handleDescOp.get();
+
+			/* verify that descriptor can be resolved */
+			MethodHandle resolvedHandle;
+			try {
+				resolvedHandle = (MethodHandle)handleDesc.resolveConstantDesc(resolveLookup);
+			} catch(ReflectiveOperationException e) {
+				Assert.fail(testName + ": ReflectiveOperationException thrown at MethodHandleDesc.resolveConstantDesc() call");
+				return;
+			}
+
+			String originalString = handle.type().descriptorString();
+			String newString = resolvedHandle.type().descriptorString();
+			logger.debug(testName + ": Descriptor of original MethodHandle " + originalString + " descriptor of MethodHandleDesc is: " + newString);
```

Comment:
We don't need to convert to Strings here - the MethodType objects can be directly compared.